src/main.c

```diff
@@ -807,14 +859,26 @@ static void handler_MonomeGridKey(s32 data) {
 					keyfirst_pos = x;
 				}
 				else if(key_alt == 1) {
-					if(x == LENGTH)
+					if(x == LENGTH) {
 						w.wp[pattern].step_mode = mForward;
-					else if(x == LENGTH-1)
+                    }
```

Comment:
Just to make sure we’re seeing the same thing:
![screen shot 2016-10-25 at 11 25 48 am](https://cloud.githubusercontent.com/assets/1838637/19692405/d5527e54-9aa5-11e6-96e9-286bba6cf0e9.png)

Also appears to be off on lines 310-343